service/storage_proxy.cc

```diff
@@ -1192,16 +1192,12 @@ future<> paxos_response_handler::learn_decision(paxos::proposal decision, bool a
 
         if (_proxy->get_cdc_service()->needs_cdc_augmentation(update_mut_vec)) {
             f_cdc = _proxy->get_cdc_service()->augment_mutation_call(_timeout, std::move(update_mut_vec), tr_state)
-                    .then([this, base_tbl_id] (std::vector<mutation>&& mutations) {
+                    .then([this, base_tbl_id] (std::tuple<std::vector<mutation>, cdc::augmentation_metadata>&& t) {
+                auto mutations = std::move(std::get<0>(t));
+                auto meta = std::move(std::get<1>(t));
                 // Pick only the CDC ("augmenting") mutations
-                mutations.erase(std::remove_if(mutations.begin(), mutations.end(), [base_tbl_id = std::move(base_tbl_id)] (const mutation& v) {
-                    return v.schema()->id() == base_tbl_id;
-                }), mutations.end());
-                if (mutations.empty()) {
-                    return make_ready_future<>();
-                }
-
-                return _proxy->mutate_internal(std::move(mutations), _cl_for_learn, false, tr_state, _permit, _timeout);
+                auto range = boost::make_iterator_range(mutations.end() - meta.total_cdc_mutations, mutations.end());
```

Comment:
Assuming the order in mutations is bad. Let's keep the old code which was safer.

Reply:
Done, new version does not depend on the order.